Automattic/jetpack
# Contact From: Newsletter integration error handling

### Refactor the newsletter integration component and add an error state if the plugin if the user doesn't have access to the plugins: (based on these comments
https://github.com/Automattic/jetpack/pull/16902#pullrequestreview-473661000

#### Fixes #
* Adding Error state if the user doesn't have access to the plugins
* Some minor copy changes
* More consistent usage of the different Button variations

#### Changes proposed in this Pull Request:
N/A

#### Jetpack product discussion
N/A

#### Does this pull request change what data or activity we track or use?
No

#### Testing instructions:

## test case 1
* Go to Posts > Add New
* Search for ‘Newsletter Sign-up’.
* Select the form block in the Gutenberg editor.
* Collapse the "Newsletter Integration" on the right side
<img width="303" alt="Screenshot 2020-08-25 at 11 07 56" src="https://user-images.githubusercontent.com/47211808/91158448-ed74f680-e6c6-11ea-9e8d-d339ef6e052c.png">
* Click install plugin creative mail plugin

![image](https://user-images.githubusercontent.com/47211808/91158563-1b5a3b00-e6c7-11ea-9c44-ec77a1385640.png)

* Click open creative mail, this should open the creative mail plugin in a new tab.

## test case 2
* Make sure that the Creative Mail plugin is *installed but deactivated*
* Go to Posts > Add New
* Search for ‘Newsletter Sign-up’.
* Select the form block in the Gutenberg editor.
* Collapse the "Newsletter Integration" on the right side
![image](https://user-images.githubusercontent.com/47211808/91158676-4e043380-e6c7-11ea-9aca-5f14ef6a6c1c.png)

* Click active plugin
![image](https://user-images.githubusercontent.com/47211808/90694105-577d3e00-e278-11ea-8672-0cd3f70452ae.png)
* Click open creative mail, this should open the creative mail plugin in a new tab.

## test case 3
* Make sure that you break your WP instance, (no write access on the plugin folder)
* Go to Posts > Add New
* Search for ‘Newsletter Sign-up’.
* Select the form block in the Gutenberg editor.
* Collapse the "Newsletter Integration" on the right side
![image](https://user-images.githubusercontent.com/47211808/91158773-6ffdb600-e6c7-11ea-85a3-8562c3bdb28e.png)
* Click install plugin
![image](https://user-images.githubusercontent.com/47211808/91158958-b18e6100-e6c7-11ea-8766-24651abaaf39.png)




#### Proposed changelog entry for your changes:
* TBD
